tubular/base.py

```diff
@@ -64,6 +66,13 @@ def __init__(
             msg = f"{self.classname()}: verbose must be a bool"
             raise TypeError(msg)
 
+        if copy is not None:
+            warnings.warn(
+                "copy argument no longer used and will be deprecated in a future release",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
```

Comment:
S: Would a check on init for `pd.options.mode.copy_on_write` being set to `True` be of use? I doubt many users want the old behaviour (which will actually become unsupported behaviour come Pandas 3.0 - 'default and only mode' is the line).

Reply:
Test for copy on write added [here](https://github.com/lvgig/tubular/pull/197/commits/4657f8cbeda7b61ced5f38bb6dc6b7c1077d4d90)

Reply:
Dropped test [here](https://github.com/lvgig/tubular/pull/197/commits/fc56bc00cc194420a96fa7acc6f85f5789f4eed1) decided not to add warning to base for this behaviour as very unlikely to set argument to False by accident.

Reply:
Agreed - if a user knows to change the copy-on-write setting, they know enough to burn their own fingers. Plus, it's a setting that is pandas specific, so wouldn't be active in a abstracted version for #185